docs: a stub for client libraries section (#4881)
edgedb/edgedb: docs/reference/bindings/index.rst

.. _ref_bindings_overview:

================

> Client Libraries

Reviewer: ```suggestion
Client libraries
```

================

> EdgeDB client libraries are a bit higher level than usual database bindings.
> In particular, they contain:

Reviewer: ```suggestion
EdgeDB client libraries are higher level than typical database bindings. In
particular, they contain:
```

* Structured data retrieval
* Connection pooling
* Retrying of failed transactions and queries

Additionally, client libraries might provide:

* Code generation for type-safe database access

> * Query builder

Reviewer: ```suggestion
* A query builder
```
